### PR TITLE
Add flag to custom output to opt into included the default target

### PR DESCRIPTION
**Pull request by lkerford**
Custom Output typically set a custom target when generating proto classes.
Allow for custom outputs to opt into included the default target if no other target is supplied

### CODE REVIEW

File: wire-library/wire-gradle-plugin/src/test/kotlin/com/squareup/wire/gradle/WirePluginTest.kt [hunk 1]
```diff
@@ -855,6 +855,17 @@ class WirePluginTest {
...
 
+  @Test
+  fun customOutputDefaultTarget() {
+    val fixtureRoot = File("src/test/projects/custom-output-use-java-default-target")
+
+    val result = gradleRunner.runFixture(fixtureRoot) { buildAndFail() }
+    assertThat(result.task(":generateMainProtos")).isNotNull
+    assertThat(result.output)
+      .contains("Writing com.squareup.geology.Period")
+      .contains("src/test/projects/custom-output-use-java-default-target/build/generated/source/wire")
```
Comment (oldergod) [inline wire-library/wire-gradle-plugin/src/test/kotlin/com/squareup/wire/gradle/WirePluginTest.kt:866]:
```suggestion
      .contains("src/test/projects/custom-output-use-java-default-target/build/generated/source/wire".withPlatformSlashes())
```

**Review by oldergod — COMMENTED** [review]
I don't get the use case.
Why not add `java {}` along side the custom handler?

File: wire-library/wire-gradle-plugin/src/test/projects/custom-output-use-java-default-target/gradle.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+VERSION_NAME=4.0.0-SNAPSHOT
```
Comment (oldergod) [inline wire-library/wire-gradle-plugin/src/test/projects/custom-output-use-java-default-target/gradle.properties:1]: Let's delete this file.
Same for settings.gradle below

File: wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt [hunk 3]
```diff
@@ -121,6 +121,7 @@ open class CustomOutput @Inject constructor() : WireOutput() {
   var includes: List<String>? = null
   var excludes: List<String>? = null
   var exclusive: Boolean = true
+  var applyDefaultTargetIfNoTargetAssigned = false
```
Comment (oldergod) [inline wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt:124]: We need doc, or a better name. By the way, how can a custom output _not_ providing a target? Isn't it a target itself?

**Comment by lkerford** [comment]
> I don't get the use case. Why not add `java {}` along side the custom handler?

We only want to add  `java {}` if no other target has been specified. We apply the custom handler in a plugin at which time we cannot check if any target have been assigned.

File: wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt (hunk 3 above)
Comment (lkerford) [reply wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt:124]: I'm not a fan of the name, let me give it some more thought. 
So it is assumed that the customer handler generate proto classes. Our custom handler doesn't, it purely meant as addition action to be taken, after/during the generating of the proto classes. In most causes the `kotlin` or `java` target is specified but in some causes we depend on the defaults. When the rely on the defaults, the wire plugin runs the custom handler but doesn't generate any proto classes, which isn't surprising.

**Pull request closed without merging**